### PR TITLE
Feature/58 doors and levels game refactoring

### PR DESCRIPTION
**Pull request by MihailRom**
This is a refactoring solution for the "doors and levels" game.

These components were extracted:
-  phrase provider
-  input output console component
-  doors numbers generator
-  settings provider and game settings,
-  storage component

Russian is added for the second language. You can change the language using the settings file (Resources/settings.xml). For English, change the language setting to “Eng”, for Russian - to “Rus”.

Issue #58

### CODE REVIEW

File: HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/DoorsAndLevels.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+            ioComponent.WriteOutputLine(phraseProvider.GetPhrase("Start"));
+            do
+            {
+                ioComponent.WriteOutputLine(phraseProvider.GetPhrase("Intro"));
+                this.Show();
+                string resultStr = ioComponent.ReadInputLine();
+                try
+                {
+                    int result = Convert.ToInt32(resultStr);
```
[the remark this answers is not in the document]
Comment (MihailRom) [reply HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/DoorsAndLevels.cs]: Definitely! I forgot about it!

File: HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/Resources/EngLang.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<phrases>
+  <Start>Let`s start to game</Start>
+  <Intro>Choose one of the number for next level or -1 to previous level. For exit enter -2:</Intro>
```
Comment (vsatkov) [inline HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/Resources/EngLang.xml]: If you use gameSettings.exitCode and gameSettings.previousLevelCode then you need to fix it

File: HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/Resources/RusLang.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<phrases>
+  <Start>Начнем игру!</Start>
+  <Intro>Выберите одно из чисел для перехода на следующий уровень или '-1' для перехода на предыдущий. Для выхода из игры введите '-2':</Intro>
```
Comment (vsatkov) [inline HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/Resources/RusLang.xml]: If you use gameSettings.exitCode and gameSettings.previousLevelCode then you need to fix it

**Comment by sevap8** [comment]
Good afternoon, I apologize for the wait.
I see that  "StyleCop" you decide to use in future projects ?
image

**Comment by sevap8** [comment]
In the task you need to implement a menu in the console at the entrance ...
"3 Need to change some game settings using configuration file on step 1.d and check game for errors."
b) input output console component
I think it doesn't matter, but I wanted to point this out.

File: HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/Resources/RusLang.xml (hunk 3 above)
Comment (MihailRom) [reply HW2/Mikhail_Romashov/DoorsAndLevels/DoorsAndLevels/Resources/RusLang.xml]: Ok, I will fix it

**Comment by MihailRom** [comment]
@sevap8 ,
Yes, right. I will use the "StyleCop" in the next projects. Now I just installed the "StyleCop" and looked at my mistakes.

As I understood from the assignment, there is no need to enter settings at the beginning of the program in console.
If you want to change the settings, use the file settings.xml (Resources folder)

**Comment by MihailRom** [comment]
@sevap8,
Thanks for the advice! In the next projects I will do it.

**Pull request merged**